### ISSUE TITLE
Calls to clearBufferuiv and clearBufferiv are incorrect

### ISSUE THREAD

**Issue by greggman** [mrdoob/three.js#26244]
### Description

this line

https://github.com/mrdoob/three.js/blob/ed2bfb2e56ee84077372735acdf4425d96549fea/src/renderers/WebGLRenderer.js#L618

and this line

https://github.com/mrdoob/three.js/blob/ed2bfb2e56ee84077372735acdf4425d96549fea/src/renderers/WebGLRenderer.js#L626

Don't seem correct?

According to the spec: https://registry.khronos.org/webgl/specs/latest/2.0/#3.7.11

the signatures for `clearBufferiv` and  `clearBufferuiv` are

```
clearBufferiv(GLenum buffer, GLint drawbuffer, Int32List values, optional GLuint srcOffset = 0)
clearBufferuiv(GLenum buffer, GLint drawbuffer, Uint32List values, optional GLuint srcOffset = 0)
```

Notice 2nd argument `drawbuffer` is `GLint` but AFAIK that's not what's being passed in the code here. Instead `WebGLFramebuffer` object is being passed.

It's used in https://threejs.org/examples/webgl_interactive_cubes_gpu.html and strangely does not generate an error. I can only guess it's being coerced to zero and running by luck?

### Reproduction steps

1. see above

### Code

See above

### Live example

see above

### Version

dev

**Comment by gkjohnson** [comment]
Sorry - I did this in #22351 😅 I believe I thought `__webglFramebuffer` was an int handle but I see now that's not the case. I think the best thing to do here right now is just pass in `0`. However that won't allow you to clear different color attachments in an MRT texture if I understand correctly. Does three.js provide any way to clear all the attachments when rendering to MRT, though? I'm less familiar with it.

**Comment by Mugen87** [comment]
> I think the best thing to do here right now is just pass in 0.

Agreed! Let's make this change now before thinking about the MRT case.

**Comment by Mugen87** [comment]
Since https://github.com/mrdoob/three.js/pull/26249 has been merged let's consider this issue an enhancement (MRT support).

**Comment by gkjohnson** [comment]
> Since https://github.com/mrdoob/three.js/pull/26249 has been merged, let's consider this issue an enhancement (MRT support).

To continue what I mentioned above - is MRT clearing supported at all in non-integer buffer cases? Ie was it possible at all to clear MRT attachments before #22351?

**Comment by Mugen87** [comment]
> Ie was it possible at all to clear MRT attachments before https://github.com/mrdoob/three.js/pull/22351?

Looking at the code then no.

**Comment by Mugen87** [comment]
I have a clear method ready that handles MRT. I've ported the approach from `WebGPURenderer`'s WebGL 2 backend.

However, this is probably a good opportunity to focus on two open issues.

- It's currently not possible to define a clear color per attachment. The policy is right now only the first color attachment uses the background's clear color. All others get (0,0,0,1). We know this is not ideal for every use case so this should be configurable. In `WebGPURenderer`, we already have `MRTNode.setBlendMode()` to configure blending per attachment. We could introduce `MRTNode.setClearColor()` which would define the clear color per attachment. But that wouldn't work with `WebGLRenderer`.
- Do we need control which attachments are cleared when `clear()` is called? Or is it okay to always clear all color attachments?

```js
this.clear = function ( color = true, depth = true, stencil = true ) {

	let bits = 0;

	if ( color ) {

		if ( _currentRenderTarget !== null ) {

			const textures = _currentRenderTarget.textures;
			const clearColor = background.getClearColor();
			const clearAlpha = background.getClearAlpha();

			for ( let i = 0; i < textures.length; i ++ ) {

				// only the first color attachment uses the background's clear color

				const texture = textures[ i ];
				const r = ( i === 0 ) ? clearColor.r : 0;
				const g = ( i === 0 ) ? clearColor.g : 0;
				const b = ( i === 0 ) ? clearColor.b : 0;
				const a = ( i === 0 ) ? clearAlpha : 1;

				if ( INTEGER_FORMATS.has( texture.format ) ) {

					if ( UNSIGNED_TYPES.has( texture.type ) ) {

						uintClearColor[ 0 ] = r;
						uintClearColor[ 1 ] = g;
						uintClearColor[ 2 ] = b;
						uintClearColor[ 3 ] = a;
						_gl.clearBufferuiv( _gl.COLOR, i, uintClearColor );

					} else {

						intClearColor[ 0 ] = r;
						intClearColor[ 1 ] = g;
						intClearColor[ 2 ] = b;
						intClearColor[ 3 ] = a;
						_gl.clearBufferiv( _gl.COLOR, i, intClearColor );

					}

				} else {

					if ( premultipliedAlpha ) {

						floatClearColor[ 0 ] = r * a;
						floatClearColor[ 1 ] = g * a;
						floatClearColor[ 2 ] = b * a;
						floatClearColor[ 3 ] = a;

					} else {

						floatClearColor[ 0 ] = r;
						floatClearColor[ 1 ] = g;
						floatClearColor[ 2 ] = b;
						floatClearColor[ 3 ] = a;

					}

					_gl.clearBufferfv( _gl.COLOR, i, floatClearColor );

				}

			}

		} else {

			bits |= _gl.COLOR_BUFFER_BIT;

		}

	}

	if ( depth ) {

		bits |= _gl.DEPTH_BUFFER_BIT;
		this.state.buffers.depth.setMask( true );

	}

	if ( stencil ) {

		bits |= _gl.STENCIL_BUFFER_BIT;
		this.state.buffers.stencil.setMask( 0xffffffff );

	}

	if ( bits !== 0 ) {

		_gl.clear( bits );

	}

};
```
